x/wasm/internal/keeper/keeper.go

```diff
@@ -82,7 +82,7 @@ func NewKeeper(cdc *codec.Codec, storeKey sdk.StoreKey, paramSpace params.Subspa
 		authZPolicy:   DefaultAuthorizationPolicy{},
 		paramSpace:    paramSpace,
 	}
-	keeper.queryPlugins = DefaultQueryPlugins(bankKeeper, stakingKeeper, keeper).Merge(customPlugins)
+	keeper.queryPlugins = DefaultQueryPlugins(bankKeeper, stakingKeeper, &keeper).Merge(customPlugins)
```

Comment:
Nice to use a pointer/reference here. I support it.

Ah... now I get it. We copy the keeper into the query plugins before we set `keeper.queryPlugins` :exploding_head: :facepalm: 

Now I understand how this ends up as nil and all the issues.

Thanks for hunting it down, I was super busy today with other org stuff.

Reply:
Something that may or may not be a bug, similarly to this, is that all the methods on the keeper take it by value. I think you should review the codebase for this sort of thing, because there may be many more places where we copy things that should be passed by reference.

Reply:
This is all over the cosmos sdk. I remember a large debate with the core devs when someone wanted to make the keepers pointers. There was big backlash on the grounds of immuntabilitt or such. Anyway, I agree with you, I prefer pointer methods